File: DSA/Manav-Rajput/Day41_question2.java
class Solution {
    public int sumSubseqWidths(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        long mod = 1_000_000_007;
        long[] powerOfTwo = new long[n];
        
        powerOfTwo[0] = 1;
        for (int i = 1; i < n; i++) {
            powerOfTwo[i] = (powerOfTwo[i - 1] * 2) % mod;
        }
        
        long sum = 0;
        for (int i = 0; i < n; i++) {
            sum = (sum + nums[i] * (powerOfTwo[i] - powerOfTwo[n - 1 - i]) % mod) % mod;
        }
        
        return (int)((sum + mod) % mod);
    }
}

File: DSA/Arshdeep-Kaur/Day 41/Day41_Question2.cpp
class Solution {
public:
    int sumSubseqWidths(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        long c = 1, res = 0, mod = 1e9 + 7, n = nums.size();
        for (int i = 0; i < n; ++i, c = c * 2 % mod) {
            res = (res + nums[i] * c - nums[n - i - 1] * c) % mod;
        }

        return (res + mod) % mod;
    }
};
